src/nvim/window.c

```diff
@@ -6662,6 +6652,32 @@ static void last_status_rec(frame_T *fr, bool statusline, bool is_stl_global)
   }
 }
 
+// Add or remove window bars from windows depending on the value of 'winbar'.
+void set_winbar(void)
+{
+  FOR_ALL_WINDOWS_IN_TAB(wp, curtab) {
+    if (*p_wbr != NUL || *wp->w_p_wbr != NUL) {
+      if (wp->w_winbar_height) {
+        continue;
+      }
+      wp->w_winrow_off += 1;
+      wp->w_height_inner -= 1;
```

Comment:
this feels a little hackish. Instead of adjusting dependent variables around, perhaps we can move the calculation of `w_winrow_off` to `win_set_inner_size` and just call it whenever the value changed. Then we don't need two branches here heither.